bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -164,10 +240,33 @@ public void setReconnectStrategy(AbstractReconnectStrategy reconnectStrategy) {
         reconnectStrategy.setBrokerConnection(this);
     }
 
+    /**
+     * @return Return the reconnect strategy
+     */
     public @Nullable AbstractReconnectStrategy getReconnectStrategy() {
         return this.reconnectStrategy;
     }
 
+    /**
+     * Set a timeout executor. If none is set, you will not be notified of connection timeouts, this
+     * also includes a non-firing reconnect strategy. The default executor is none.
+     *
+     * @param executor One timer will be created when a connection attempt happens
+     * @param timeoutInMS Timeout in milliseconds
+     */
+    public void setTimeoutExecutor(@Nullable ScheduledExecutorService executor, int timeoutInMS) {
+        timeoutExecutor = executor;
+        this.timeout = timeoutInMS;
+    }
+
+    public @Nullable Executor getTimeoutExecutor() {
```

Comment:
Do we really need that getters? What is the intention? The executor should not be of any interest after it has been set and we should prevent from reuse the executor of an instance by a consumer for something different.